packages/vite/src/node/plugins/html.ts

```diff
@@ -1154,23 +1177,42 @@ export function preImportMapHook(
  * Move importmap before the first module script and modulepreload link
  */
 export function postImportMapHook(): IndexHtmlTransformHook {
-  return (html) => {
+  return async (html, { path }) => {
+    if (!importMapRE.test(html)) return
     if (!importMapAppendRE.test(html)) return
+    let importMapAppendIndex = 0
+    const s = new MagicString(html)
+    await traverseHtml(html, path, (node) => {
+      if (!nodeIsElement(node)) {
+        return
+      }
 
-    let importMap: string | undefined
-    html = html.replace(importMapRE, (match) => {
-      importMap = match
-      return ''
+      const { nodeName, attrs, sourceCodeLocation } = node
+      if (
+        importMapAppendIndex &&
+        nodeName === 'script' &&
+        attrs.some((attr) => attr.name === 'type' && attr.value === 'importmap')
+      ) {
+        s.move(
+          sourceCodeLocation!.startTag!.startOffset,
+          sourceCodeLocation!.endOffset,
+          importMapAppendIndex,
+        )
```

Comment:
I guess this doesn't work if importmap exists before `<script type=module>` or `<link rel=modulepreload>`. I think we need to store the position here and call `s.move` after the `traverseHtml` call.

Reply:
If this is the case, importmap does not seem to need to be moved. 🤔

Reply:
Ah, you're right. Please forget about this.